src/coreclr/jit/importercalls.cpp

```diff
@@ -3176,6 +3179,43 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                             retNode = gtNewIconNode(
                                 (info.compCompHnd->getClassAttribs(hClass) & CORINFO_FLG_BYREF_LIKE) ? 1 : 0);
                             break;
+                        case NI_System_Type_get_IsPrimitive:
+                            retNode = gtNewFalse();
+
+                            // Since asCorInfoType returns underlying types for enums, we need to check
```

Comment:
There are other weird corner cases like `RuntimeArgumentHandle`. I think `asCorInfoType` returns `CORINFO_TYPE_NATIVEINT` for those as well.

Reply:
Ah I see, looks like `getTypeForPrimitiveNumericClass` is what I need, but it ignores CHAR and BOOL, I guess I can change the JIT-EE api by adding `bool* isBlittable` to `getTypeForPrimitiveNumericClass`

Reply:
@jkotas do you think it's fine to keep as is? all JIT usages of `getTypeForPrimitiveNumericClass` are checking return result explicitly anyway so they're fine with it returning char and bool too.

I can probably rename it to remove "Numeric" from the name.

Reply:
> adding bool* isBlittable to getTypeForPrimitiveNumericClass

Blittability is interop concept. It does not sound right to use it to infer whether the type is primitive (for reflection definition of primitive).

> remove "Numeric" from the name so it will be getTypeForPrimitiveClass

There is existing `getTypeForPrimitiveValueClass`. The distinction between `getTypeForPrimitiveClass` vs. `getTypeForPrimitiveValueClass` would be very subtle and non-obvious from the name. I think `getTypeForPrimitiveNumericClass` is fine as is.

If you would like to start with reusing or overloading the existing APIs, I think `getTypeForPrimitiveValueClass` is probably the best one to start with. Either call `getTypeForPrimitiveValueClass` and then do some kind of check of enums; or add `bool* isActualPrimitive = nullptr` argument to `getTypeForPrimitiveValueClass`.

Reply:
Changed to `getTypeForPrimitiveValueClass ` + `isEnum` check